pkg/syncer/syncer.go

```diff
@@ -445,20 +453,41 @@ func (c *Controller) process(ctx context.Context, h holder) error {
 // - if the object is a configmap it handles the "kube-root-ca.crt" name mapping
 // - if the object is a serviceaccount it handles the "default" name mapping
 func transformName(syncedObject *unstructured.Unstructured, direction SyncDirection) {
+	configMapGVR := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"}
+	serviceAccountGVR := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ServiceAccount"}
+	secretGVR := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"}
+
 	switch direction {
 	case SyncDown:
-		if syncedObject.GetKind() == "ConfigMap" && syncedObject.GetName() == "kube-root-ca.crt" {
+		if syncedObject.GroupVersionKind() == configMapGVR && syncedObject.GetName() == "kube-root-ca.crt" {
 			syncedObject.SetName("kcp-root-ca.crt")
 		}
-		if syncedObject.GetKind() == "ServiceAccount" && syncedObject.GetName() == "default" {
+		if syncedObject.GroupVersionKind() == serviceAccountGVR && syncedObject.GetName() == "default" {
 			syncedObject.SetName("kcp-default")
 		}
+		// TODO(jmprusi): We are rewriting the name of the object into a non random one so we can reference it from the deployment transformer
+		//                but this means that means than more than one default-token-XXXX object will overwrite the same "kcp-default-token"
+		//				  object. This must be fixed.
+		if syncedObject.GroupVersionKind() == secretGVR && strings.Contains(syncedObject.GetName(), "default-token-") {
+			syncedObject.SetName("kcp-default-token")
+		}
 	case SyncUp:
-		if syncedObject.GetKind() == "ConfigMap" && syncedObject.GetName() == "kcp-root-ca.crt" {
+		if syncedObject.GroupVersionKind() == configMapGVR && syncedObject.GetName() == "kcp-root-ca.crt" {
 			syncedObject.SetName("kube-root-ca.crt")
 		}
-		if syncedObject.GetKind() == "ServiceAccount" && syncedObject.GetName() == "kcp-default" {
+		if syncedObject.GroupVersionKind() == serviceAccountGVR && syncedObject.GetName() == "kcp-default" {
 			syncedObject.SetName("default")
 		}
 	}
 }
+
+func getDefaultMutators(from *rest.Config) mutatorGvrMap {
+	mutatorsMap := make(mutatorGvrMap)
+
+	deploymentMutator := mutators.NewDeploymentMutator(from)
+	secretMutator := mutators.NewSecretMutator()
+
+	mutatorsMap[deploymentMutator.GVR()] = deploymentMutator.Mutate
+	mutatorsMap[secretMutator.GVR()] = secretMutator.Mutate
```

Comment:
nit: why the multiple steps, just `mutatorsMap[deploymentMutator.GVR()] = mutators.NewDeploymentMutator(from)`

Reply:
deploymentMutator.GVR() is a method of the deployment mutator, so I need to instantiate it first, I can change it to a single line, but readability suffers a little bit:

```
mutatorsMap[mutators.NewDeploymentMutator(from).GVR()] = mutators.NewDeploymentMutator(from).Mutate
```

Reply:
(No action required) Maybe GVR doesn't have to be provided by a mutator type method? It's static data, it could just be a var in the same namespace (e.g. DeploymentGVR).